protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -806,13 +815,46 @@ func (rpccs *RPCConsumerServer) relayInner(ctx context.Context, singleConsumerSe
 		relaySentTime := time.Now()
 		connectCtx, connectCtxCancel := context.WithTimeout(ctx, relayTimeout)
 		metadataAdd := metadata.New(map[string]string{common.IP_FORWARDING_HEADER_NAME: consumerToken})
+		metadataAdd.Append(common.LAVA_LB_UNIQUE_ID_HEADER, singleConsumerSession.EndpointConnection.GetLbUniqueId())
+		utils.LavaFormatTrace("Sending relay to provider",
+			utils.LogAttr("GUID", ctx),
+			utils.LogAttr("lbUniqueId", singleConsumerSession.EndpointConnection.GetLbUniqueId()),
+		)
 		connectCtx = metadata.NewOutgoingContext(connectCtx, metadataAdd)
 		defer connectCtxCancel()
 
 		// add consumer processing timestamp before provider metric and start measuring time after the provider replied
 		rpccs.rpcConsumerLogs.AddMetricForProcessingLatencyBeforeProvider(analytics, rpccs.listenEndpoint.ChainID, rpccs.listenEndpoint.ApiInterface)
 
 		reply, err = endpointClient.Relay(connectCtx, relayRequest, grpc.Trailer(&relayResult.ProviderTrailer))
+
+		providerUniqueId := relayResult.ProviderTrailer.Get(chainlib.RpcProviderUniqueIdHeader)
+		if len(providerUniqueId) > 0 {
+			utils.LavaFormatTrace("Received provider unique id",
+				utils.LogAttr("GUID", ctx),
+				utils.LogAttr("provider", relayRequest.RelaySession.Provider),
+				utils.LogAttr("providerUniqueId", providerUniqueId),
+			)
+
+			if !singleConsumerSession.VerifyProviderUniqueId(providerUniqueId[0]) {
+				return reply, 0, utils.LavaFormatError("provider unique id mismatch",
+					errors.Join(lavasession.SessionOutOfSyncError, lavasession.BlockEndpointError),
```

Comment:
i want a unitest with this flow, catching errors can be tricky

Reply:
Add the following commits to fix this:
8c2777135bde9c66f017b9997a8487248672cf8c
fe7a1959d9d6fcb1122a7499d192921152c09c9a
df03af315f6a7c98e32f4fe31ab5ab4e1c3cfede